src/services/shop/shop-router.test.ts

```diff
@@ -0,0 +1,431 @@
+        const response = await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        expect(JSON.parse(response.text)).toMatchObject({
+            ...TESTER_SHOP_ORDER,
+            quantity: [3],
+        });
+
+        const updatedOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(updatedOrder?.items).toContain(TESTER_SHOP_ITEM.itemId);
+        expect(updatedOrder?.quantity[0]).toBe(3);
+    });
+
+    it("increases quantity when adding existing item to cart", async () => {
+        // First addition
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        // Second addition
+        const response = await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        expect(JSON.parse(response.text)).toMatchObject({
+            ...TESTER_SHOP_ORDER,
+            quantity: [4],
+        });
+
+        const updatedOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(updatedOrder?.items).toContain(TESTER_SHOP_ITEM.itemId);
+        expect(updatedOrder?.quantity[0]).toBe(4);
+    });
+
+    it("returns NotFound for non-existent item", async () => {
+        await postAsAttendee("/shop/cart/non-existent-item").expect(StatusCode.ClientErrorNotFound);
+    });
+
+    it("returns BadRequest when insufficient shop quantity", async () => {
+        await Models.ShopItem.create({
+            ...TESTER_SHOP_ITEM,
+            itemId: "out-of-stock-item",
+            quantity: 0,
+        });
+
+        await postAsAttendee("/shop/cart/out-of-stock-item").expect(StatusCode.ClientErrorBadRequest);
+    });
+
+    it("returns PaymentRequired when insufficient points", async () => {
+        await Models.ShopItem.create({
+            ...TESTER_SHOP_ITEM,
+            itemId: "expensive-item",
+            price: 2000,
+            quantity: 1,
+        });
+
+        await postAsAttendee("/shop/cart/expensive-item").expect(StatusCode.ClientErrorPaymentRequired);
+    });
+
+    it("creates new cart if user doesn't have one", async () => {
+        // Delete any existing cart
+        await Models.ShopOrder.deleteOne({ userId: TESTER_PROFILE.userId });
+
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        const newOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(newOrder).toBeTruthy();
+        expect(newOrder?.items).toContain(TESTER_SHOP_ITEM.itemId);
+        expect(newOrder?.quantity[0]).toBe(1);
+    });
+});
+
+describe("DELETE /shop/cart/:itemId", () => {
+    it("allows user to remove an item from the cart", async () => {
+        // Add item to the cart first
+        await Models.ShopOrder.updateOne({ userId: TESTER_PROFILE.userId }, { items: ["test-item-1"], quantity: [0] });
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
```

Comment:
this depends on the other endpoint. While it is good to test both of these in one test, you should have at least one test that does not rely on the other endpoint. Otherwise, both of these tests will fail.